packages/app/src/components/DynamicRoot/DynamicRoot.tsx

```diff
@@ -0,0 +1,336 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import { createApp } from '@backstage/app-defaults';
+import { AppConfig } from '@backstage/config';
+import { BackstageApp } from '@backstage/core-app-api';
+import { AnyApiFactory, BackstagePlugin } from '@backstage/core-plugin-api';
+
+import { AppsConfig, getScalprum } from '@scalprum/core';
+import { useScalprum } from '@scalprum/react-core';
+
+import bindAppRoutes from '../../utils/dynamicUI/bindAppRoutes';
+import extractDynamicConfig, {
+  configIfToCallable,
+} from '../../utils/dynamicUI/extractDynamicConfig';
+import initializeRemotePlugins from '../../utils/dynamicUI/initializeRemotePlugins';
+import defaultThemes from './defaultThemes';
+import DynamicRootContext, {
+  ComponentRegistry,
+  DynamicRootContextValue,
+  RemotePlugins,
+  ScalprumMountPoint,
+  ScalprumMountPointConfig,
+} from './DynamicRootContext';
+import Loader from './Loader';
+
+export type StaticPlugins = Record<
+  string,
+  {
+    plugin: BackstagePlugin;
+    module:
+      | React.ComponentType<{}>
+      | { [importName: string]: React.ComponentType<{}> };
+  }
+>;
+
+type EntityTabMap = Record<string, { title: string; mountPoint: string }>;
+
+export const DynamicRoot = ({
```

Comment:
I would not recommend going down this path of replicating the dynamic plugin implementation code in this repo.  A better approach would be to look into basing the e2e tests on using the backstage-showcase container image instead.